atc/worker/pool.go

```diff
@@ -41,7 +47,29 @@ type Pool interface {
 		ContainerSpec,
```

Comment:
Can we remove `SelectWorker` from the interface now that `WaitForWorker` is the entrypoint?